test/dsp.jl

```diff
@@ -58,3 +58,13 @@ a = [1., 2., 1., 2.]
 b = [1., 2., 3.]
 @test conv(a, b) ≈ [1., 4., 8., 10., 7., 6.]
 @test conv(complex.(a, ones(4)), complex(b)) ≈ complex.([1., 4., 8., 10., 7., 6.], [1., 3., 6., 6., 5., 3.])
+
+# Test for issue #188: deconv mutates inputs
+let b = [4.0, 2.0, 1.0]; a = [2.0, 1.0]
```

Comment:
Can you guard the test with `if VERSION >= v"0.7.0-DEV.602"`? Prior to that, the functions from `Base` are used which contain this bug, too, so the test fails.